src/sentry/static/sentry/app/views/organizationIncidents/details/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import LoadingIndicator from 'app/components/loadingIndicator';
+import LoadingError from 'app/components/loadingError';
+import {PageContent} from 'app/styles/organization';
+import withApi from 'app/utils/withApi';
+
+import IncidentHeader from './header';
+import Incidents from './incidents';
+
+class OrganizationIncidentDetails extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {isLoading: false, hasError: false};
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    this.setState({isLoading: true, hasError: false});
+    const {
+      api,
+      params: {orgId, incidentId},
+    } = this.props;
+
+    api
+      .requestPromise(`/organizations/${orgId}/incidents/${incidentId}/`)
+      .then(incident => {
+        this.setState({incident, isLoading: false, hasError: false});
+      })
+      .catch(() => {
+        this.setState({isLoading: false, hasError: true});
+      });
```

Comment:
Probably should discuss some of this in frontend TSC, but I've been enjoying writing these using the async fashion:

```js
try {
    const incident = await api.requestPromise(`/organizations/${orgId}/incidents/${incidentId}/`)
    this.setState({incident});
} catch (error) {
    this.setState({hasError: error});
}

this.setState({loading: false})
```

Reply:
I would be hesitant to change the pattern we use everywhere just in one place